source/util/FileUtils.cpp

```diff
@@ -302,3 +302,25 @@ bool FileUtils::FileExists(const string &filename)
     ifstream f(expandedPath);
     return f.good();
 }
+
+bool FileUtils::IsValidFilePath(const string &filePath)
+{
+    wordexp_t word;
+    switch (wordexp(filePath.c_str(), &word, 0))
+    {
+        case 0:
+            break;
+        case WRDE_BADCHAR:
```

Comment:
replace this with `WRDE_NOSPACE` and have a default case to return false so all other failure cases will be covered. 

```
case 0: 
    break;
case WRDE_NOSPACE:
      wordfree (&word);
default:                    /* Some other error.  */
      return false;
```